--- ldetect-0.13.4.new/pci.cpp	2013-11-18 09:07:10.477781584 +0000
+++ ldetect-0.13.4/pci.cpp	2013-11-18 09:03:19.758769448 +0000
@@ -5,6 +5,7 @@
 #include <dirent.h>
 #include <pci/header.h>
 #include <libkmod.h>
+#include <unistd.h>
 
 #include "common.h"
 #include "pci.h"
@@ -148,7 +149,21 @@
 	    devname << hexFmt(e.pci_domain, 4, false) << ":" <<  hexFmt(e.bus, 2, false) <<
 		":" << hexFmt(e.pciusb_device, 2, false) << "." << hexFmt(e.pci_function, 0, false);
 
-	    std::ifstream f(std::string("/sys/bus/pci/devices/").append(devname.str()).append("/modalias").c_str());
+            std::string sysDir = std::string("/sys/bus/pci/devices/").append(devname.str());
+            char buf[1024];
+            ssize_t n = readlink(std::string(sysDir + "/driver").c_str(), buf, sizeof(buf));
+            if(n < 0)
+              buf[0] = 0;
+            else 
+              buf[n] = 0;
+
+            char* drv;
+            if ((drv = strrchr(buf, '/')))
+              e.module = drv +1;
+            else
+              e.module = buf;
+
+	    std::ifstream f(sysDir.append("/modalias").c_str());
 	    if (f.is_open()) {
 		std::string modalias;
 		getline(f, modalias);
